src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs

```diff
@@ -231,58 +233,48 @@ private static (Dictionary<StringSpanOrdinalKey, int>?, Dictionary<int, string>?
+        /// Encode text to a list of tokens.
         /// </summary>
-        /// <param name="text">The text to encode. If the value of the parameter <paramref name="isSpecialToken"/> is true, the entire text will be treated as a special token.</param>
-        /// <param name="isSpecialToken">Specifies whether the entire <paramref name="text"/> is considered a special token.</param>
+        /// <param name="text">The text to encode.</param>
         /// <returns>The list of tokens generated from the text tokenization.</returns>
-        public override IReadOnlyList<Token> Encode(string text, bool isSpecialToken = false)
+        public override IReadOnlyList<Token> Encode(ReadOnlySpan<char> text)
         {
             Token[] tokens;
 
-            if (string.IsNullOrEmpty(text))
+            if (text.IsEmpty)
             {
                 return Array.Empty<Token>();
             }
 
-            if (isSpecialToken)
+            if (_cache.TryGetValue(text, out (int[] Bytes, string Token) ids))
             {
-                if (_specialTokensEncoder?.TryGetValue(text, out int id) is true)
-                {
-                    return new List<Token> { new(id, text, (0, text.Length)) };
-                }
-
-                throw new InvalidOperationException($"The special token {text} doesn't exist in the tokenizer");
-            }
-
-            if (_cache.TryGetValue(text, out int[]? ids))
-            {
-                tokens = new Token[ids.Length];
-                tokens[0] = new Token(ids[0], text, (0, text.Length));
-                for (int i = 1; i < ids.Length; i++)
+                tokens = new Token[ids.Bytes.Length];
+                tokens[0] = new Token(ids.Bytes[0], ids.Token, (0, ids.Token.Length));
+                for (int i = 1; i < ids.Bytes.Length; i++)
                 {
                     // One word split mapped to multiple Ids. Make the offset of the remaining token point at the end with zero width.
-                    tokens[i] = new Token(ids[i], "", (text.Length, 0));
+                    tokens[i] = new Token(ids.Bytes[i], "", (text.Length, 0));
                 }
 
                 return tokens;
             }
 
             // cache miss
-            if (_vocab.TryGetValue(text, out int mappedId))
+            if (_vocab.TryGetValue(text, out (int Id, string Token) mappedId))
             {
-                return new Token[1] { new(mappedId, text, (0, text.Length)) };
+                return new Token[1] { new(mappedId.Id, mappedId.Token, (0, mappedId.Token.Length)) };
             }
 
             byte[] arrayPoolArray = ArrayPool<byte>.Shared.Rent(Encoding.UTF8.GetMaxByteCount(text.Length));
-            int encodedLength = Helpers.GetUtf8Bytes(text.AsSpan(), arrayPoolArray);
+            int encodedLength = Helpers.GetUtf8Bytes(text, arrayPoolArray);
 
             int[] encodedIds = BytePairEncoder.BytePairEncode(arrayPoolArray.AsMemory(0, encodedLength), _encoder);
```

Comment:
Can the arrayPoolArray be returned after this line instead of at the end?

Reply:
I moved it as suggested.